homeassistant/components/wwlln/geo_location.py

```diff
@@ -0,0 +1,217 @@
+DEFAULT_ICON = 'mdi:flash'
+DEFAULT_UPDATE_INTERVAL = timedelta(minutes=5)
+
+SIGNAL_DELETE_ENTITY = 'delete_entity_{0}'
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up WWLLN based on a config entry."""
+    client = hass.data[DOMAIN][DATA_CLIENT][entry.entry_id]
+    manager = WWLLNEventManager(
+        hass,
+        async_add_entities,
+        client,
+        entry.data[CONF_LATITUDE],
+        entry.data[CONF_LONGITUDE],
+        entry.data[CONF_RADIUS],
+        entry.data[CONF_WINDOW],
+        entry.data[CONF_UNIT_SYSTEM])
+    await manager.async_init()
+
+
+class WWLLNEventManager:
+    """Define a class to handle WWLLN events."""
+
+    def __init__(
+            self,
+            hass,
+            async_add_entities,
+            client,
+            latitude,
+            longitude,
+            radius,
+            window_seconds,
+            unit_system):
+        """Initialize."""
+        self._async_add_entities = async_add_entities
+        self._client = client
+        self._hass = hass
+        self._latitude = latitude
+        self._longitude = longitude
+        self._managed_strike_ids = set()
+        self._radius = radius
+        self._strikes = {}
+        self._window_seconds = window_seconds
+
+        self._unit_system = unit_system
+        if unit_system == CONF_UNIT_SYSTEM_IMPERIAL:
+            self._unit = LENGTH_MILES
+        else:
+            self._unit = LENGTH_KILOMETERS
+
+    @callback
+    def _create_events(self, ids_to_create):
+        """Create new geo location events."""
+        events = []
+        for strike_id in ids_to_create:
+            strike = self._strikes[strike_id]
+            event = WWLLNEvent(
+                strike['distance'],
+                strike['lat'],
+                strike['long'],
+                self._unit,
+                strike_id,
+                strike['unixTime'])
+            events.append(event)
+
+        self._async_add_entities(events)
+
+    @callback
+    def _remove_events(self, ids_to_remove):
+        """Remove old geo location events."""
+        for strike_id in ids_to_remove:
+            async_dispatcher_send(
+                self._hass, SIGNAL_DELETE_ENTITY.format(strike_id))
+
+    async def async_init(self):
+        """Schedule regular updates based on configured time interval."""
+        async def update(event_time):
+            """Update."""
+            await self.async_update()
+
+        await self.async_update()
+        async_track_time_interval(self._hass, update, DEFAULT_UPDATE_INTERVAL)
+
+    async def async_update(self):
+        """Refresh data."""
+        _LOGGER.debug('Refreshing WWLLN data')
+
+        try:
+            self._strikes = await self._client.within_radius(
+                self._latitude,
+                self._longitude,
+                self._radius,
+                unit=self._unit_system,
+                window=timedelta(seconds=self._window_seconds))
```

Comment:
If we need a timedelta, why not store the window as timedelta? The option is timedelta from the beginning, right?

Reply:
Timedelta's aren't JSON-serializable, so they can't be stored in as-is in the config entry (thus, my workaround: store it as seconds, then transform  that back to a timedelta [since that's what `aiowwlln` needs]).

Does that make sense?

Reply:
Makes perfect sense. Thanks for clarifying!